awsagentprovider/src/main/java/software/amazon/opentelemetry/javaagent/providers/AwsMetricAttributeGenerator.java

```diff
@@ -539,6 +561,61 @@ private static void setRemoteResourceTypeAndIdentifier(SpanData span, Attributes
     }
   }
 
+  private static void setRemoteResourceAccessKeyAndRegion(
+      SpanData span, AttributesBuilder builder) {
+    if (isKeyPresent(span, AWS_REMOTE_RESOURCE_ACCESS_KEY)) {
+      String remoteAccessKey = span.getAttributes().get(AWS_REMOTE_RESOURCE_ACCESS_KEY);
+      builder.put(AWS_REMOTE_RESOURCE_ACCESS_KEY, remoteAccessKey);
+    }
+
+    if (isKeyPresent(span, AWS_REMOTE_RESOURCE_REGION)) {
+      String remoteRegion = span.getAttributes().get(AWS_REMOTE_RESOURCE_REGION);
+      builder.put(AWS_REMOTE_RESOURCE_REGION, remoteRegion);
+    }
+  }
+
+  private static void setRemoteResourceAccountIdAndRegion(
+      SpanData span, AttributesBuilder builder) {
+    String remoteAccountId = UNKNOWN_REMOTE_ACCOUNT_ID;
+    String remoteRegion = UNKNOWN_REMOTE_REGION;
+
+    if (isKeyPresent(span, AWS_QUEUE_URL)) {
+      try {
+        // queue url format: https://sqs.{region}.amazonaws.com/{account_id}/{queue_name}
+        String queueUrl = span.getAttributes().get(AWS_QUEUE_URL);
+        queueUrl = queueUrl.replace("https://", "");
+
+        String[] parts = queueUrl.split("/");
+        remoteAccountId = parts[1];
+
+        String domain = parts[0];
+        String[] domainParts = domain.split("\\.");
+        remoteRegion = domainParts[1];
+      } catch (Exception e) {
+        logger.log(
+            Level.WARNING,
+            "Remote resource account id and region cannot be populated due to invalid url",
+            e);
+      }
+    } else {
+      for (AttributeKey<String> attributeKey : ARN_ATTRIBUTES) {
+        if (isKeyPresent(span, attributeKey)) {
+          String arn = span.getAttributes().get(attributeKey);
+          remoteAccountId = arn.split(":")[4];
+          remoteRegion = arn.split(":")[3];
+          break;
+        }
+      }
+    }
+
+    if (!remoteAccountId.equals(UNKNOWN_REMOTE_ACCOUNT_ID)
```

Comment:
Why not just set the default values to null or empty string? It seems you are not using UNKNOWN anywhere as assigned values.